primitives/s2s/src/lib.rs

```diff
@@ -0,0 +1,168 @@
+// This file is part of Darwinia.
+//
+// Copyright (C) 2018-2021 Darwinia Network
+// SPDX-License-Identifier: GPL-3.0
+//
+// Darwinia is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Darwinia is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Darwinia. If not, see <https://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+// --- paritytech ---
+use sp_core::H160;
+use sp_std::{vec, vec::Vec};
+// --- darwinia-network ---
+use codec::{Decode, Encode};
+use dp_asset::token::Token;
+use dp_contract::mapping_token_factory::s2s::S2sRemoteUnlockInfo;
+
+/// The parameters box for the pallet runtime call.
+#[derive(Encode, Decode, Debug, PartialEq, Eq, Clone)]
+pub enum CallParams<AccountId> {
+	#[codec(index = 0)]
+	S2sIssuingPalletRegisterFromRemote(Token),
+	#[codec(index = 1)]
+	S2sIssuingPalletIssueFromRemote(Token, H160),
+	#[codec(index = 2)]
+	S2sBackingPalletUnlockFromRemote(AccountId, S2sRemoteUnlockInfo),
+}
```

Comment:
we must add the prefix XXXPallet to indicate which pallet this remote call belongs. It seems not good.

Reply:
I already added `S2sBackingPallet` or `S2sIssuingingPallet`. Or i am missing something?

Reply:
What if two call from two separate pallet have the same call index?

Reply:
Good question. Just give the actual index. No conflict, a demo like this:

```rs
use codec::Encode;

#[derive(Encode)]
enum T {
	#[codec(index = 1)]
	A,
	#[codec(index = 1)]
	B,
}
fn main() {
	assert_eq!(T::A.encode(), vec![1]);
	assert_eq!(T::B.encode(), vec![1]);
}
```

Reply:
it's a little confusing to put methods in different pallets together.